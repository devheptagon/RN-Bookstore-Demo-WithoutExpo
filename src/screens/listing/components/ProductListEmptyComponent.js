import React from 'react'
import PropTypes from 'prop-types'
import { Text, StyleSheet } from 'react-native'
import Layout from '../../../constants/Layout'
import FlexView from '../../components/FlexView'
import { ProductListingTypes } from '../../../constants/Enums'
import Texts from '../../../constants/Texts'

export const ProductListEmptyComponent = props => {
    return <FlexView centerAll style={styles.Wrapper}>
        <Text>
        { 
            props.listingType === ProductListingTypes.Basket
                ? Texts.EmptyBasketText
                : Texts.EmptyListText
        }
        </Text>
    </FlexView>
}

const styles = StyleSheet.create({
    Wrapper: {
        height: Layout.window.height * 0.8
    }
})

ProductListEmptyComponent.propTypes = {  
    listingType: PropTypes.number
}

export default ProductListEmptyComponent