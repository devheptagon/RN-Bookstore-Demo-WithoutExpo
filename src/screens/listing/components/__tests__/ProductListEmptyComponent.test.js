import React from 'react';
import {ProductListEmptyComponent} from '../ProductListEmptyComponent';
import renderer from 'react-test-renderer';

test('ProductListEmptyComponent renders correctly', () => {
  const tree = renderer.create(<ProductListEmptyComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});