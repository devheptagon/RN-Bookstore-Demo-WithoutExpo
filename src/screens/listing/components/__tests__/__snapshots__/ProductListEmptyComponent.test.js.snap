// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListEmptyComponent renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "height": 1067.2,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Nothing to list yet!
  </Text>
</View>
`;
